### DISCUSSION TITLE
This completion box does not look right

### DISCUSSION THREAD

**Discussion by azegas** [company-mode/company-mode#1475] [Q&A] [answered]
https://github.com/company-mode/company-mode/assets/78803192/dee94ae9-4523-4d73-bd3d-03e40368122f

Launched emacs with 'emacs -q'. The company suggestions looks somehow off. I would expect to see something like his:

![image](https://github.com/company-mode/company-mode/assets/78803192/7d8b8af6-ac86-4013-ad21-46c1f657f093)

But the graphics are completely off as you could see.

Running emacs on nixos wsl.

Curious to hear what you think about this issue. Is it Emacs issue? Is it company issue? Is it WSL issue? Is it nixos issue?

Thanks!

**Comment by azegas** [comment]
@terlar perhaps you might have an idea? :)

**Reply by terlar** [reply]
Looks like issue related to variable width. You need to use a monospaced font in order to have correct display with company or use something like company-box which doesn't display things in the main buffer.

**Reply by dgutov** [reply]
Yes, the situation is that you (@azegas) seem to be using a proportional font. A monospaced one should fix it.

But you could also try using company-mode from the master branch (also published to MELPA), which should deal with different character sizes much better now. See the entry in NEWS.md which is currently 7th from the top.

**Reply by azegas** [reply]
![image](https://github.com/company-mode/company-mode/assets/78803192/a2d28fdf-c044-44b6-8f69-2a1bcd9f0269)

when I launch emacs with `emacs -q`, the problem persists. `emacs -q` should not load any modifications from my `init.el`, right(not including any possible font modifications)?

Will try to look into how to use a monospaced font.

Also will check the newer branch like you suggest @dgutov .

**Reply by azegas** [reply]
btw if I turn off company and run corfu instead, it looks like this:
![image](https://github.com/company-mode/company-mode/assets/78803192/647d3bb0-dcbe-4ff2-b295-1ad7d58b14e0)

**Reply by dgutov** [reply]
I guess for some reason your Emacs defaults to a non-monospaced font? Maybe none of the monospaced fonts it knows about are available in your system? Or something like that.

Corfu starts out with a frontend that works like `company-box` (recommended above) or `company-posframe` (works better on some systems). That mechanism has both certain advantages and downsides too.

**Reply by azegas** [reply]
I checked my emacs config on another machine, same misalignment happens.

Now thinking it could actually be the problem on my side, in my system. On both machines I am running on quite lightweight nixos system and there are **other** parts of emacs that are not aligned as it should be (or at least how I remember it used to be aligned when I was running emacs on ubuntu repo).

**pdb** help commands are not aligned(assume they should be in nice columns):
![image](https://github.com/company-mode/company-mode/assets/78803192/96008a11-0f23-4510-8899-15b6b8b29099)

Commands are not aligned, definitely should be in nice distinct columns:
![image](https://github.com/company-mode/company-mode/assets/78803192/83b40581-52f8-45eb-bc81-f951cb304588)

**Agenda view** commands are not aligned, definitely should be in nice distinct columns:

![image](https://github.com/company-mode/company-mode/assets/78803192/0cd9af05-f79c-4665-bc56-91c34d088a5a)

**Reply by azegas** [reply]
Perhaps I should try to add different font(monospaced) to my system. Currently it's like such:

![image](https://github.com/company-mode/company-mode/assets/78803192/46ef1857-c97d-4053-b1b3-c7b1155b6dbd)

**Comment by azegas** [comment, marked as answer]
Indeed, installing a monospaced font, Iosevka or FiraCode on my system - did the trick:

Company displays as intended now:
![image](https://github.com/company-mode/company-mode/assets/78803192/d2f3a447-aea8-4d4f-8efd-85a7518d2bf3)

Other views got fixed as a consequence also!:
![image](https://github.com/company-mode/company-mode/assets/78803192/834f5f87-ac56-4b7d-904c-263ee897d95a)
![image](https://github.com/company-mode/company-mode/assets/78803192/760cb2b4-e0f1-442a-93fa-be5d87156331)
![image](https://github.com/company-mode/company-mode/assets/78803192/67b33ba6-786a-4b30-859a-255aab1c4663)

Great, thank you!

**Reply by dgutov** [reply]
Congrats! Iosevka and FiraCode are both pretty nice fonts, too. Looking good.
